repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasRelationshipStoreV2.java

```diff
@@ -369,28 +380,29 @@ public void deleteById(String guid, boolean forceDelete) throws AtlasBaseExcepti
             }
         }
 
-        sendNotifications(entityRetriever.mapEdgeToAtlasRelationship(edge), OperationType.RELATIONSHIP_DELETE);
+        if (deleteDelegate.getHandler() instanceof SoftDeleteHandlerV1)
+            onRelationshipsMutated(RequestContext.get().getRelationshipMutationMap());
+        else
+            onRelationshipsMutated(RequestContext.get().getRelationshipMutationMap());
 
         if (LOG.isDebugEnabled()) {
             LOG.debug("<== deleteById({}): {}", guid);
         }
     }
 
-
-
     @Override
     public AtlasEdge getOrCreate(AtlasVertex end1Vertex, AtlasVertex end2Vertex, AtlasRelationship relationship) throws AtlasBaseException {
         AtlasEdge ret = getRelationship(end1Vertex, end2Vertex, relationship);
 
         if (ret == null) {
             ret = createRelationship(end1Vertex, end2Vertex, relationship, false);
+            recordRelationshipMutation(RelationshipMutation.RELATIONSHIP_CREATE, ret, entityRetriever);
```

Comment:
We are only recording here and not calling - onRelationshipsMutated?

Reply:
It is being called in AtlasEntityStoreV2 after the entity notifications.